app/controllers/groups_controller.rb

```diff
@@ -29,11 +32,32 @@ def update
     end
   end
 
+  def new
+    @parent_groups = Group.all
+                          .includes(:children)
+                          .reject { |g| g.children.empty? }
+                          .sort_by { |g| g.children.length }
+                          .reverse
```

Comment:
After checking the data integrity for the resorts in the database, it can be simplified to:
```ruby
@resorts = Group.type_resort
```

Reply:
I like how simple the query became.  Should I write a rake task, for setting the resorts?

Reply:
I think it's not worth it, it is a one time job and relies on the current data in the db which is not so exact. In this case it's better and faster do a manual check.

Reply:
Should it go in like this, or we refactor it after the db integrity check?

Reply:
It can go here, first of all this is a new piece of code and I think the one you wrote depends on the database integrity too, so both way it's worth a quick check.